src/azul/service/hca_response_v5.py

```diff
@@ -298,9 +308,53 @@ def make_projects(self, entry):
                 translated_project['insdcProjectAccessions'] = project.get('insdc_project_accessions', [None])
                 translated_project['insdcStudyAccessions'] = project.get('insdc_study_accessions', [None])
                 translated_project['supplementaryLinks'] = project.get('supplementary_links', [None])
+                translated_project['contributorMatrices'] = self.make_contributor_matrices_tree(entry)
             projects.append(translated_project)
         return projects
 
+    def make_contributor_matrices_tree(self, entry) -> JSON:
+        """
+        Returns a stratification tree for the Contributor-generated matrix files
+        in the entry.
+        """
+        contributor_matrices = entry['contents']['contributor_matrices']
+        if contributor_matrices:
+            files = one(contributor_matrices)['file']
+        else:
+            return {}
+        unique_values = defaultdict(set)  # Unique values per stratification key
+        for file in files:
+            # Transform the stratification string into a list of dicts (1 per line)
+            file['stratification'] = [
+                {kv[0]: kv[1] for kv in (pair.split('=') for pair in s.split(';'))}
+                for s in file['stratification'].split('\n')
+            ]
+            for k, v in [(k, v) for d in file['stratification'] for k, v in d.items()]:
+                unique_values[k].add(v)
+        # To produce a tree with the most shared base branches possible we sort
+        # the stratum keys ascending by the count of unique values per key
+        sorted_keys = sorted(unique_values, key=lambda k: len(unique_values[k]))
+        tree = {}
+        for file in files:
+            file_url = furl(config.service_endpoint(),
+                            path=f"/fetch/repository/files/{file['uuid']}",
```

Comment:
Should exploit that `path` can be a list or tuple, to eliminate the need for string interpolation and to utilize the automatic escaping of special characters in path elements.